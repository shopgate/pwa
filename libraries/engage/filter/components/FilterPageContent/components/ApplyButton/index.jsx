import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { I18n, Button, SurroundPortals } from '@shopgate/engage/components';
import { PORTAL_FILTER_APPLY_BUTTON } from '@shopgate/engage/filter/constants';
import { withWidgetSettings } from '@shopgate/engage/core';
import styles from './style';

/**
 * The filter apply button component.
 * @returns {JSX}
 */
const FilterApplyButton = ({ active, onClick, widgetSettings }) => {
  const { buttonTextColor, buttonTextColorDisabled } = widgetSettings;
  const buttonColor = active ? buttonTextColor : buttonTextColorDisabled;

  return (
    <SurroundPortals
      portalName={PORTAL_FILTER_APPLY_BUTTON}
      portalProps={{ active, onClick, widgetSettings }}
    >
      <div className={styles.wrapper}>
        <Button
          className={styles.button}
          flat
          type="primary"
          onClick={onClick}
          disabled={!active}
          testId="applyFilterButton"
        >
          <I18n.Text string="filter.apply" style={{ color: buttonColor }} />
        </Button>
      </div>
    </SurroundPortals>
  );
};

FilterApplyButton.propTypes = {
  active: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  widgetSettings: PropTypes.shape().isRequired,
};

export default withWidgetSettings(memo(FilterApplyButton), '@shopgate/engage/components/AppBar');
