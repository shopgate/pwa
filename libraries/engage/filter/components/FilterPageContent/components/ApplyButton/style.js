import { css } from 'glamor';

const wrapper = css({
  marginTop: 10,
});

const button = css({
  padding: '0 !important',
}).toString();

export default {
  wrapper,
  button,
};
