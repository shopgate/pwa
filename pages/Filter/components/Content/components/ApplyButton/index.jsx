import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import Button from '@shopgate/pwa-ui-shared/Button';
import styles from './style';

/**
 * The filter apply button component.
 */
class FilterApplyButton extends PureComponent {
  static propTypes = {
    active: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
  };

  /**
   * @returns {JSX}
   */
  render() {
    const { active, onClick } = this.props;

    return (
      <div className={styles}>
        <Button flat type="primary" onClick={onClick} disabled={!active} testId="applyFilterButton">
          <I18n.Text string="filter.apply" />
        </Button>
      </div>
    );
  }
}

export default FilterApplyButton;
