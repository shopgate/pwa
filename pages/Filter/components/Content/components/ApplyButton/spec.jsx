import React from 'react';
import { shallow } from 'enzyme';
import ApplyButton from './index';

const clickMock = jest.fn();

jest.mock('@shopgate/engage/core', () => ({
  withWidgetSettings: function withWidgetSettings(Comp) {
    return props => (<Comp widgetSettings={{}} {...props} />);
  },
}));

describe('Filter: <ApplyButton />', () => {
  it('should render as activated', () => {
    const wrapper = shallow(<ApplyButton active onClick={() => { }} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render as deactivated', () => {
    const wrapper = shallow(<ApplyButton active={false} onClick={() => { }} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle clicks', () => {
    const wrapper = shallow(<ApplyButton onClick={clickMock} active />).dive();
    expect(wrapper).toMatchSnapshot();
    wrapper.find('Button').simulate('click');
    expect(clickMock).toHaveBeenCalled();
  });
});
