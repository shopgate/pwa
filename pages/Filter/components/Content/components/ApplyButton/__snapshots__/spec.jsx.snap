// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter: <ApplyButton /> should handle clicks 1`] = `
<div
  className={
    Object {
      "data-css-prcfqy": "",
    }
  }
>
  <Button
    className=""
    disabled={false}
    flat={true}
    onClick={[MockFunction]}
    testId="applyFilterButton"
    type="primary"
    wrapContent={true}
  >
    <Translate
      className=""
      params={Object {}}
      string="filter.apply"
    />
  </Button>
</div>
`;

exports[`Filter: <ApplyButton /> should render as activated 1`] = `
<div
  className={
    Object {
      "data-css-prcfqy": "",
    }
  }
>
  <Button
    className=""
    disabled={false}
    flat={true}
    onClick={[Function]}
    testId="applyFilterButton"
    type="primary"
    wrapContent={true}
  >
    <Translate
      className=""
      params={Object {}}
      string="filter.apply"
    />
  </Button>
</div>
`;

exports[`Filter: <ApplyButton /> should render as deactivated 1`] = `
<div
  className={
    Object {
      "data-css-prcfqy": "",
    }
  }
>
  <Button
    className=""
    disabled={true}
    flat={true}
    onClick={[Function]}
    testId="applyFilterButton"
    type="primary"
    wrapContent={true}
  >
    <Translate
      className=""
      params={Object {}}
      string="filter.apply"
    />
  </Button>
</div>
`;
